File: palindrome_checker.py
def is_palindrome(s):
    # Clean input: ignore non-alphanumeric and case
    s = ''.join(c.lower() for c in s if c.isalnum())
    return s == s[::-1]

if __name__ == "__main__":
    import sys
    if len(sys.argv) > 1:
        input_str = sys.argv[1]
        if is_palindrome(input_str):
            print("Palindrome")
            exit(0)
        else:
            print("Not a palindrome")
            exit(1)
    else:
        print("Usage: python palindrome_checker.py <string>")
        exit(2)

File: PalindromeChecker.java
public class PalindromeChecker {
    public static boolean isPalindrome(String text) {
        int left = 0;
        int right = text.length() - 1;
        while (left < right) {
            if (text.charAt(left) != text.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        if (args.length > 0) {
            String input = args[0];
            if (isPalindrome(input)) {
                System.out.println(input + " is a palindrome");
                System.exit(0);
            } else {
                System.out.println(input + " is not a palindrome");
                System.exit(1);
            }
        } else {
            System.out.println("Usage: java PalindromeChecker <string>");
            System.exit(2);
        }
    }
}
